### ISSUE TITLE
Unable to open trace files when profiler is built in RelWithDebInfo

### ISSUE THREAD

**Issue by slomp** [wolfpld/tracy#903]
In the current master, I saved a large trace to a file, and later when I tried to reopen it, I got the following:
![image](https://github.com/user-attachments/assets/68e07b21-230a-4591-8ade-59d66601e08f)

I noticed this behavior on Windows/Intel.

**Comment by wolfpld** [comment]
What's the hex dump of first 4 bytes?

**Comment by slomp** [comment]
74 72 FD 50

**Comment by wolfpld** [comment]
That looks to be garbled uncompressed data stream (`TracyWorker.cpp:584`). Can you check if any of the initial data blocks (`FileRead::GetNextDataBlock()`) contains the expected header (`FileHeader`)?

**Comment by slomp** [comment]
Sure, I'll try to get a debug build later to check.

I forgot to mention that I used Zstd with compression level 3, and concurrency 4.
With the non-Zstd modes, the trace file opens as expected.

**Comment by slomp** [comment]
Update: it works when I build profiler in the Debug configuration.
It's probably some uninitialized variable somewhere. I'll try to find it.

**Comment by slomp** [comment]
`ZSTD_decompressStream()` returns code `0xff...ffec` (-20) which reads: "Frame requires too much memory for decoding"

**Comment by slomp** [comment]
Bizarre... The error only occurs in `RelWithDebInfo` builds...
Debug and Release work just fine.
Maybe ZSTD has problems when it gets built with that configuration?

**Comment by wolfpld** [comment]
Hmm... There is a similar problem with emscripten, where some traces fail to load when set as initial, but load correctly when another trace is loaded first and then closed. One of the ways this can manifest itself is runaway memory consumption. This is probably related.

**Comment by slomp** [comment]
Yeah, I'll try to step through the `ZSTD_decompressStream` again later.

(I just tried opening a non-zstd trace file first, close it, and then open the zstd trace file, and it also fails.)

**Comment by slomp** [comment]
I also believe this is a Windows-only thing (could be MSVC related, even).
(I don't recall having trouble with Zstd trace files in RelWithDebInfo on Mac and Linux, but I'll double-check later.)

**Comment by jkriegshauser** [comment]
Just came here to report this  as we're also running into it

**Comment by slomp** [comment]
Doesn't look like this is happening anymore, so closing for now.
